### PR TITLE
Fix type error in initializing TensorFlow computation graph for inference.

### PR DESCRIPTION
**Pull request by dwadden**
@nfoti and I discovered a bug when doing inference in a neural net with a softmax final layer (e.g. for multiclass classification). This PR fixes it.

### CODE REVIEW

File: edward/inferences/inference.py [hunk 1]
```diff
@@ -132,8 +132,15 @@ def __init__(self, latent_vars=None, data=None, model_wrapper=None):
               raise TypeError("Observed variable bindings do not have same "
                               "shape.")
 
-            # If value is a np.ndarray, store it in the graph.
-            ph = tf.placeholder(tf.float32, value.shape)
```
Comment (dwadden) [inline edward/inferences/inference.py]: The TensorFlow placeholder needs to have the same basic type (e.g. integer or float) as the numpy array for which it's serving as a placeholder.

File: tests/test-inferences/test_integer_variables.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                  sigma=softplus(tf.Variable(tf.random_normal([D, n_hidden]))))
+    qW_1 = Normal(mu=tf.Variable(tf.random_normal([n_hidden, K])),
+                  sigma=softplus(tf.Variable(tf.random_normal([n_hidden, K]))))
+    qb_0 = Normal(mu=tf.Variable(tf.random_normal([n_hidden])),
+                  sigma=softplus(tf.Variable(tf.random_normal([n_hidden]))))
+    qb_1 = Normal(mu=tf.Variable(tf.random_normal([K])),
+                  sigma=softplus(tf.Variable(tf.random_normal([K]))))
+    inference = ed.KLqp({W_0: qW_0, b_0: qb_0, W_1: qW_1, b_1: qb_1},
+                        data={y: y_train})
+    inference.run()
```
Comment (dwadden) [inline tests/test-inferences/test_integer_variables.py]: Without the patch in this PR, this line fails with error:
`typeError: DataType float32 for attr 'Tlabels' not in list of allowed values: int32, int64`

**Comment by dustinvtran** [comment]
perfect. this also fixes #407. i ran it locally and it looks great. (i think you were right to not also do the casting at line 156 when `isinstance(key, str)`. this is the case for model wrappers, and in their implementation we force everything to be `float32` by convention.)

not sure how to interpret that build error.

**Comment by dwadden** [comment]
Huh yeah it works locally for me as well... weird build error. I'll take a look.

**Comment by dustinvtran** [comment]
man, tensorflow is weird. the unit test succeeds if it's run on the file separately, but fails if it also runs `test_bayesian_nn.py`. the failure is always whichever one is run second. i'm not sure what sort of artifact is remaining in one graph vs the other.

**Comment by dwadden** [comment]
Haha well that's not good. Do you think there's something I can do on my end to fix it?

**Comment by dustinvtran** [comment]
i spotted the fix. it looks like if you add the line `sess = tf.InteractiveSession()` to the top of `test_integer_variables.py`, the test passes. the error arises when the session in `test_integer_variables.py` tries to use the session built in `test_bayesian_nn.py`; resetting the default session by calling `tf.InteractiveSession()` breaks this interaction.

i recommending adding `sess = tf.InteractiveSession()` to the top of the `test_integer(self)` method (i.e., line 23), and for consistency to do the same for `test_bayesian_nn.py`.

**Comment by dwadden** [comment]
Nice. OK, will do.

**Comment by dwadden** [comment]
OK, looks like things work now.
